server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -0,0 +1,468 @@
+        return readerManager;
+    }
+
+    @Override
+    public boolean isTranslogSyncNeeded() {
+        return translog.syncNeeded();
+    }
+
+    @Override
+    public boolean ensureTranslogSynced(Stream<Translog.Location> locations) throws IOException {
+        boolean synced = translog.ensureSynced(locations);
+        if (synced) {
+            translog.trimUnreferencedReaders();
+        }
+        return synced;
+    }
+
+    @Override
+    public void syncTranslog() throws IOException {
+        translog.sync();
+        translog.trimUnreferencedReaders();
+    }
+
+    @Override
+    public Closeable acquireHistoryRetentionLock() {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    @Override
+    public Translog.Snapshot newChangesSnapshot(
+        String source,
+        long fromSeqNo,
+        long toSeqNo,
+        boolean requiredFullRange,
+        boolean accurateCount
+    ) throws IOException {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    @Override
+    public int countNumberOfHistoryOperations(String source, long fromSeqNo, long toSeqNumber) throws IOException {
+        return 0;
+    }
+
+    @Override
+    public boolean hasCompleteOperationHistory(String reason, long startingSeqNo) {
+        return false;
+    }
+
+    @Override
+    public long getMinRetainedSeqNo() {
+        return localCheckpointTracker.getProcessedCheckpoint();
+    }
+
+    @Override
+    public TranslogStats getTranslogStats() {
+        return translog.stats();
+    }
+
+    @Override
+    public Translog.Location getTranslogLastWriteLocation() {
+        return translog.getLastWriteLocation();
+    }
+
+    @Override
+    public long getPersistedLocalCheckpoint() {
+        return localCheckpointTracker.getPersistedCheckpoint();
+    }
+
+    public long getProcessedLocalCheckpoint() {
+        return localCheckpointTracker.getProcessedCheckpoint();
+    }
+
+    @Override
+    public SeqNoStats getSeqNoStats(long globalCheckpoint) {
+        return localCheckpointTracker.getStats(globalCheckpoint);
+    }
+
+    @Override
+    public long getLastSyncedGlobalCheckpoint() {
+        return translog.getLastSyncedGlobalCheckpoint();
+    }
+
+    @Override
+    public long getIndexBufferRAMBytesUsed() {
+        return 0;
+    }
+
+    @Override
+    public List<Segment> segments(boolean verbose) {
+        return Arrays.asList(getSegmentInfo(getLatestSegmentInfos(), verbose));
+    }
+
+    @Override
+    public void refresh(String source) throws EngineException {}
+
+    @Override
+    public boolean maybeRefresh(String source) throws EngineException {
+        return false;
+    }
+
+    @Override
+    public void writeIndexingBuffer() throws EngineException {}
+
+    @Override
+    public boolean shouldPeriodicallyFlush() {
+        return false;
+    }
+
+    @Override
+    public void flush(boolean force, boolean waitIfOngoing) throws EngineException {}
```

Comment:
We would need an implementation to ensure translog rolls over, add a TODO?

Reply:
Good catch this is missing.  I think this would happen when a replica receives new segments belonging to a new commit point, indicating the primary has flushed, will add a test.

Reply:
updated to roll xlog when new commit point is received.